TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metadata.py

```diff
@@ -134,80 +134,56 @@ def _init_from_fs_snapshot(self, snapshot: FileSystemSnapshot) -> None:
-                ])
-                password_field_index = len(command) - 1
+                command_args.update({
+                    "--target-username": self._target_cluster.auth_details.get("username"),
+                    "--target-password": self._target_cluster.get_basic_auth_password()
+                })
                 logger.info("Using basic auth for target cluster")
             except KeyError as e:
                 raise ValueError(f"Missing required auth details for target cluster: {e}")
 
         if self._target_cluster.allow_insecure:
-            command.append("--target-insecure")
+            command_args.update({"--target-insecure": None})
 
         if self._index_allowlist:
-            command.extend(["--index-allowlist", ",".join(self._index_allowlist)])
+            command_args.update({"--index-allowlist": ",".join(self._index_allowlist)})
 
         if self._index_template_allowlist:
-            command.extend(["--index-template-allowlist", ",".join(self._index_template_allowlist)])
+            command_args.update({"--index-template-allowlist": ",".join(self._index_template_allowlist)})
 
         if self._component_template_allowlist:
-            command.extend(["--component-template-allowlist", ",".join(self._component_template_allowlist)])
+            command_args.update({"--component-template-allowlist": ",".join(self._component_template_allowlist)})
 
         if self._otel_endpoint:
-            command.extend(["--otel-collector-endpoint", self._otel_endpoint])
+            command_args.update({"--otel-collector-endpoint": self._otel_endpoint})
 
-        if password_field_index:
-            display_command = command[:password_field_index] + ["********"] + command[password_field_index:]
-        else:
-            display_command = command
-        logger.info(f"Migrating metadata with command: {' '.join(display_command)}")
-
-        if detached_log:
-            return self._run_as_detached_process(command, detached_log)
-        return self._run_as_synchronous_process(command)
-
-    def _run_as_synchronous_process(self, command) -> CommandResult:
-        try:
-            # Pass None to stdout and stderr to not capture output and show in terminal
-            subprocess.run(command, stdout=None, stderr=None, text=True, check=True)
-            logger.info(f"Metadata migration for snapshot {self._snapshot_name} completed")
-            return CommandResult(success=True, value="Metadata migration completed")
-        except subprocess.CalledProcessError as e:
-            logger.error(f"Failed to migrate metadata: {str(e)}")
-            return CommandResult(success=False, value=f"Failed to migrate metadata: {str(e)}")
-
-    def _run_as_detached_process(self, command, log_file) -> CommandResult:
+        command_runner = CommandRunner(command_base, command_args,
+                                       password_field="--target-password",
```

Comment:
nit: in the future i can see metadata migrate needing access to source cluster in which case password_field can be renamed sensitive fields and be a list